src/java/org/apache/cassandra/audit/BinAuditLogger.java

```diff
@@ -40,7 +41,7 @@
 
     private volatile BinLog binLog;
 
-    public BinAuditLogger()
+    public BinAuditLogger(Map<String, String> params)
```

Comment:
instead of adding this (most often) unused parameter, could we just grab the params from `DatabaseDescriptor.getAuditLoggingOptions().logger.parameters` in the constructor if we need them?

Reply:
@krummas we indeed technically could but I am missing the point ... for other cases this is way how to do things and now we are introducing something "custom". So lets remove this parameter from all other cases it occurs? Having it inconsistent is even worse imho.

Reply:
@krummas please check second commit if you wish to go with that option. "other cases" I meant e.g. `SeedProvider`, its constructor accepts this map, even empty ... `ICompressor` implementations have static method accepting that map and so on.

Reply:
yeah looks like we do this in other places, fine to leave as is